src/mon/OSDMonitor.cc

```diff
@@ -14629,6 +14629,25 @@ int OSDMonitor::_prepare_remove_pool(
       }
     }
   }
+  // remove any old pg_upmap_primary mapping for this pool
+  for (auto& p : osdmap.pg_upmap_primaries) {
+    if (p.first.pool() == pool) {
+      dout(10) << __func__ << " " << pool
+               << " removing obsolete pg_upmap_primaries " << p.first
+               << dendl;
+      pending_inc.old_pg_upmap_primary.insert(p.first);
+    }
+  }
+
+  // remove any new pg_upmap_primary mapping for this pool
```

Comment:
I would change the comment to "remove any **pending** pg_upmap_primary ..."

Reply:
done